src/components/Truncate/Truncate.tsx

```diff
@@ -34,16 +35,20 @@ export class Truncate extends React.PureComponent<
   }
 
   componentDidMount() {
-    if (this.props.type === 'auto') {
-      this.setState({
-        isTruncated: this.isTruncated(this.props),
-      })
-    }
+    this._isMounted = true
+    setTimeout(() => {
+      if (this.props.type === 'auto' && this._isMounted) {
+        this.setState({
+          isTruncated: this.isTruncated(this.props),
+        })
```

Comment:
One thing we often do for situations like this is create a method called `safeSetState` that wraps `setState` with an `if (this._isMounted) {}` check. So in this case something like:

```
if (this.props.type === 'auto') {
  this.safeSetState({ isTruncated: this.isTruncated(this.props) })
}
```

Not a blocker but could be a nice refactor.